packages/react-internal/src/components/Modal/Modal.base.tsx

```diff
@@ -177,63 +107,313 @@ export class ModalBase extends React.Component<IModalProps, IDialogState> implem
+              break;
+            }
+            case KeyCodes.left: {
+              setCoordinates(prevValue => ({ x: getClampedAxis('x', prevValue.x - delta), y: prevValue.y }));
+              break;
+            }
+            case KeyCodes.right: {
+              setCoordinates(prevValue => ({ x: getClampedAxis('x', prevValue.x + delta), y: prevValue.y }));
+              break;
+            }
+            default: {
+              handledEvent = false;
+            }
+          }
+          if (handledEvent) {
+            ev.preventDefault();
+            ev.stopPropagation();
+          }
+        }
+      };
+
+      internalState.lastSetCoordinates = coordinates;
+      setModalMenuClose();
+      internalState.isInKeyboardMoveMode = true;
+
+      internalState.events.on(win, 'keydown', handleKeyDown, true /* useCapture */);
+      internalState.disposeOnKeyDown = () => {
+        internalState.events.off(win, 'keydown', handleKeyDown, true /* useCapture */);
+        internalState.disposeOnKeyDown = undefined;
+      };
+    };
+
+    const handleExitKeyboardMoveMode = () => {
+      internalState.lastSetCoordinates = ZERO;
+      internalState.isInKeyboardMoveMode = false;
+      internalState.disposeOnKeyDown?.();
+    };
+
+    const registerForKeyUp = (): void => {
+      const handleKeyUp = (ev: React.KeyboardEvent<HTMLElement>): void => {
+        // Needs to handle the CTRL + ALT + SPACE key during keyup due to FireFox bug:
+        // https://bugzilla.mozilla.org/show_bug.cgi?id=1220143
+        if (ev.altKey && ev.ctrlKey && ev.keyCode === KeyCodes.space) {
+          if (elementContains(internalState.scrollableContent, ev.target as HTMLElement)) {
+            toggleModalMenuOpen();
+            ev.preventDefault();
+            ev.stopPropagation();
+          }
+        }
+      };
+
+      if (!internalState.disposeOnKeyUp) {
+        internalState.events.on(win, 'keyup', handleKeyUp, true /* useCapture */);
+        internalState.disposeOnKeyUp = () => {
+          internalState.events.off(win, 'keyup', handleKeyUp, true /* useCapture */);
+          internalState.disposeOnKeyUp = undefined;
+        };
+      }
+    };
+
+    React.useEffect(() => {
+      clearTimeout(internalState.onModalCloseTimer);
+      // Opening the dialog
+      if (isOpen) {
+        // This must be done after the modal content has rendered
+        requestAnimationFrame(() => setTimeout(registerInitialModalPosition, 0));
+
+        setIsModalOpen(true);
+
+        // Add a keyUp handler for all key up events once the dialog is open.
+        if (dragOptions) {
+          registerForKeyUp();
+        }
+
+        internalState.hasBeenOpened = true;
+        setIsVisible(true);
+      }
+
+      // Closing the dialog
+      if (!isOpen && isModalOpen) {
+        internalState.onModalCloseTimer = setTimeout(handleModalClose, parseFloat(animationDuration) * 1000);
```

Comment:
Should this also be on a `requestAnimationFrame`?

Reply:
Discussed and we decided to leave this as-is for now.